### ISSUE TITLE
Terraform outputs sensitive information when using import block as comment

### ISSUE THREAD

**Issue by kderck** [hashicorp/terraform#33544]
### Terraform Version

```shell
Terraform v1.5.2
```

### Terraform Configuration Files

```terraform
import {
  to = aws_iam_access_key.access_key_1
  id = "AK..." # Key value omitted
}
```

Will output 

```terraform
# generated_resources.tf
# __generated__ by Terraform from "AK..."
resource "aws_iam_access_key" "access_key" {
  pgp_key = null
  status  = "Active"
  user    = "..."
}
```

### Debug Output

N/A

### Expected Behavior

Terraform maybe should hash the value instead of plain text. While  having only the AWS Access Key ID without the Secret Access Key is not as dangerous as having both, but it is still essential to treat the Access Key ID as sensitive information and take measures to protect it from unauthorized access. 

### Actual Behavior

terraform creates the resources and leaks the access key ids while the import blocks can be moved.

### Steps to Reproduce

terraform plan -generate-config-out=generated_resources.tf

### References

https://developer.hashicorp.com/terraform/language/import/generating-configuration

**Comment by kmoe** [comment]
Thanks, I agree we should be conservative here and not write the import id to the comment in case it's sensitive.

**Comment by kmoe** [comment]
Yes, "sensitive" is not accurate here. It is sufficient that the comment is unnecessary.

**Comment by apparentlymart** [comment]
We currently prohibit using a sensitive map (in the sense defined by the language runtime) in resource and module `for_each` so that we can avoid having resource identifiers that we can't show in the UI.

That might be sufficient precedent to also prohibit using a sensitive string as the `id` for import, although if we can avoid actually showing the ID string in the UI anywhere (aside from showing the real resource attributes that just happen to be derived from it, where our UI code already respects sensitivity) then that's less of a concern.

**Comment by brental** [comment]
The import block also outputs the import ID in output as where the resource is being imported from and as the Id of the resource in the plan before apply. This means for cases like the venafi_certificate resource where the import ID contains the password for the private key that sensitive information gets exposed. This occurs even if the sensitive function is used on the value assigned to the id field of the import block.

The terraform import command outputs the import ID as well.

This means that there is no way to import resources such as venafi_certificate without exposing sensitive information.

Here is the doco on the import ID venafi_certificate:
https://registry.terraform.io/providers/Venafi/venafi/latest/docs/resources/venafi_certificate#import

**Comment by PFeynman** [comment]
I'm trying to import a random_password reading its value from onepassword_item and the id also appears in the log:
`module.example.random_password.password: Preparing import... [id=plain password]`
I've tried using sensitive function with the value, but it keeps showing it in plain text.
